File: playbook-java/src/main/java/io/infinitic/playbook/java/scheduler/Stop.java
package io.infinitic.playbook.java.scheduler;

import io.infinitic.clients.InfiniticClient;
import io.infinitic.playbook.java.scheduler.workflows.RecurringWorkflowScheduler;

import static io.infinitic.playbook.java.scheduler.Start.schedulerTag;

public class Stop {

    public static void main(String[] args) {
        try(InfiniticClient client = InfiniticClient.fromConfigResource("/infinitic.yml")) {
            // cancelling the existing RecurringWorkflowScheduler instances with the "scheduler" tag
            RecurringWorkflowScheduler scheduler = client.getWorkflowByTag(RecurringWorkflowScheduler.class, schedulerTag);
            client.cancel(scheduler);
        }
    }
}

File: playbook-kotlin/src/main/kotlin/io/infinitic/playbook/kotlin/scheduler/Stop.kt
package io.infinitic.playbook.kotlin.scheduler

import io.infinitic.clients.InfiniticClient
import io.infinitic.playbook.kotlin.scheduler.workflows.RecurringWorkflowScheduler

fun main() {
    InfiniticClient.fromConfigResource("/configs/infinitic.yml").use { client ->
        // cancelling the existing RecurringWorkflowScheduler instances with the scheduler tag
        val scheduler = client.getWorkflowByTag(RecurringWorkflowScheduler::class.java, schedulerTag)
        client.cancel(scheduler)
    }
}
